elements/lisk-p2p/src/peer_pool.ts

```diff
@@ -373,18 +375,27 @@ export class PeerPool extends EventEmitter {
 		return discoveredPeers;
 	}
 
-	public selectPeersAndConnect(
+	public triggerNewConnections(
 		peers: ReadonlyArray<P2PDiscoveredPeerInfo>,
-	): ReadonlyArray<P2PDiscoveredPeerInfo> {
-		const peersToConnect = this._peerSelectForConnection({ peers });
-
-		peersToConnect.forEach((peerInfo: P2PDiscoveredPeerInfo) => {
-			const peerId = constructPeerIdFromPeerInfo(peerInfo);
+	): void {
+		const peersCount = this.getPeersCountByKind();
+		// Trigger new connections only if the maximum of outbound connections has not been reached
+		if (peersCount.outbound < this._maxOutboundConnections) {
```

Comment:
The decision for selecting peers is currently split between this code and the `this._peerSelectForConnection` function .

Ideally, the `_peerSelectForConnection` function should receive the full list of peers and return a subset of that list as output. You can pass new arguments to the `_peerSelectForConnection` function to give it more information to make the decision but the ideally the entire selection decision should happen inside `_peerSelectForConnection`.

This is because `_peerSelectForConnection` is customizable by the user of the library (e.g. our Network module) so we should give that function more control over the selection-for-connection logic.

Note that we may want to change the input arguments to the `_peerSelectForConnection` function in the future; maybe we will want to pass `_newPeers` and `_triedPeers` separately?